builtin/pack/builder.go

```diff
@@ -52,6 +52,9 @@ type BuilderConfig struct {
 
 	// Files patterns to prevent from being pulled into the build.
 	Ignore []string `hcl:"ignore,optional"`
+	
+	// Process type that will be used when setting container start command.
+	ProcessType string `hcl:"process_type,optional" default:"web"`
```

Comment:
Why default to `"web"`? I think we could set no default, update `ProcessType` to be `*string`, so if the config option is unset it just passes nil through. Then I think the `pack` package can handle the default type there instead 🤔

Reply:
> Why default to "web"?

I made `web` the default value because that's what the `pack` package [is using itself](https://github.com/buildpacks/pack/blob/216a4d1de720d51a76ccfa0d397b26150263b494/internal/build/lifecycle_execution.go#L23).

> Then I think the pack package can handle the default type there instead 🤔

The idea of letting the actual `pack` package decide the default `ProcessType` sounds good to me, but bear in mind that it does so for [`PlatformAPI` versions greater than `0.4`](https://github.com/buildpacks/pack/blob/216a4d1de720d51a76ccfa0d397b26150263b494/internal/build/lifecycle_execution.go#L380).

Reply:
Setting it to web by default here is fine but i'm unsure what is reading that struct tag to populate it. Is that a feature of HCL I'm unaware of?